### ISSUE TITLE
writeMetaDataFile crashes when a relative root Path is used

### ISSUE THREAD

**Issue by asfimport** [apache/parquet-java#1841]
In Spark, I can save an RDD to the local file system using a relative path, e.g.:

```
rdd.saveAsNewAPIHadoopFile(
        "relativeRoot",
        classOf[Void],
        tag.runtimeClass.asInstanceOf[Class[T]],
        classOf[ParquetOutputFormat[T]],
        job.getConfiguration)
```

This leads to a crash in the ParquetFileWriter.mergeFooters(..) method since the footer paths are read as fully qualified paths, but the root path is provided as a relative path:

```
org.apache.parquet.io.ParquetEncodingException: /Users/stevenshe/schema/relativeRoot/part-r-00000.snappy.parquet invalid: all the files must be contained in the root relativeRoot
```

**Reporter**: [Steven She](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stevenshe) / @stshe
**Assignee**: [Steven She](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stevenshe) / @stshe
#### PRs and other links:
- [PR #228](https://github.com/apache/parquet-mr/pull/228)

<sub>**Note**: *This issue was originally created as [PARQUET-317](https://issues.apache.org/jira/browse/PARQUET-317). Please see the [migration documentation](https://issues.apache.org/jira/browse/PARQUET-2502) for further details.*</sub>

**Comment by asfimport** [comment]
[Steven She](https://issues.apache.org/jira/browse/PARQUET-317?#comment-14599905) / @stshe:
https://github.com/apache/parquet-mr/pull/228

**Comment by asfimport** [comment]
[Ryan Blue](https://issues.apache.org/jira/browse/PARQUET-317?#comment-14602392) / @rdblue:
Merged #228. Thanks for fixing this, Steven!
